dotnet/src/webdriver/BiDi/Modules/Script/LocalValue.cs

```diff
@@ -58,86 +58,210 @@ public static LocalValue ConvertFrom(object? value)
+                return ConvertFrom(set);
 
             case IEnumerable<object?> list:
-                return new ArrayLocalValue(list.Select(ConvertFrom).ToList());
-
-            case object:
-                {
-                    const System.Reflection.BindingFlags Flags = System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Instance;
-
-                    var properties = value.GetType().GetProperties(Flags);
-
-                    var values = new List<List<LocalValue>>(properties.Length);
-                    foreach (var property in properties)
-                    {
-                        object? propertyValue;
-                        try
-                        {
-                            propertyValue = property.GetValue(value);
-                        }
-                        catch (Exception ex)
-                        {
-                            throw new BiDiException($"Could not retrieve property {property.Name} from {property.DeclaringType}", ex);
-                        }
-                        values.Add([property.Name, ConvertFrom(propertyValue)]);
-                    }
-
-                    return new ObjectLocalValue(values);
-                }
+                return ConvertFrom(list);
+
+            default:
+                return ReflectionBasedConvertFrom(value);
+        }
+    }
+
+    public static LocalValue ConvertFrom(bool? value)
+    {
+        if (value is bool b)
+        {
+            return new BooleanLocalValue(b);
+        }
+
+        return new NullLocalValue();
+    }
+
+    public static LocalValue ConvertFrom(int? value)
+    {
+        if (value is int b)
+        {
+            return new NumberLocalValue(b);
+        }
+
+        return new NullLocalValue();
+    }
+
+    public static LocalValue ConvertFrom(double? value)
+    {
+        if (value is double b)
+        {
+            return new NumberLocalValue(b);
         }
+
+        return new NullLocalValue();
+    }
+
+    public static LocalValue ConvertFrom(long? value)
+    {
+        if (value is long b)
+        {
+            return new NumberLocalValue(b);
+        }
+
+        return new NullLocalValue();
+    }
+
+    public static LocalValue ConvertFrom(string? value)
+    {
+        if (value is not null)
+        {
+            return new StringLocalValue(value);
+        }
+
+        return new NullLocalValue();
+    }
+
+    public static LocalValue ConvertFrom(DateTime? value)
+    {
+        if (value is null)
+        {
+            return new NullLocalValue();
+        }
+
+        return new DateLocalValue(value.Value.ToString("o"));
+    }
+
+    public static LocalValue ConvertFrom(BigInteger? value)
+    {
+        if (value is not null)
+        {
+            return new BigIntLocalValue(value.Value.ToString());
+        }
+
+        return new NullLocalValue();
+    }
+
+    public static LocalValue ConvertFrom(IEnumerable<object?>? value)
```

Comment:
Should be non-generic `IEnumerable`, then conversion will happen for `object` flow (via reflection).

Reply:
Resolved, but there is no non-generic `ISet`. This is why we still see `public static LocalValue ConvertFrom<T>(ISet<T?>? value)`.

Reply:
It seems that this method is unnecessary since we have `LocalValue.ConvertFrom(object)`, but from user's point of view it is good to know that `LocalValue.ConvertFrom(new HashSet<string> { });` is supported.